statev2/proto/src/lib.rs

```diff
@@ -1,28 +1,94 @@
+use mpc_stark::algebra::scalar::Scalar;
+use uuid::{Error as UuidError, Uuid};
+
+pub use protos::*;
+
 #[deny(missing_docs)]
 #[deny(clippy::missing_docs_in_private_items)]
 #[deny(unsafe_code)]
 
 /// Protobuf definitions for state transitions
+#[allow(missing_docs)]
+#[allow(clippy::missing_docs_in_private_items)]
 mod protos {
-    include!(concat!(env!("OUT_DIR"), "/state_transitions.rs"));
+    include!(concat!(env!("OUT_DIR"), "/state.rs"));
 }
 
-pub use protos::*;
+// --------------------
+// | Type Definitions |
+// --------------------
+
+/// PeerId
+impl From<String> for PeerId {
+    fn from(id: String) -> Self {
+        Self { id }
+    }
+}
+
+/// ClusterId
+impl From<String> for ClusterId {
+    fn from(id: String) -> Self {
+        Self { id }
+    }
+}
+
+/// UUID
+impl From<String> for ProtoUuid {
+    fn from(value: String) -> Self {
+        Self { value }
+    }
+}
+
+impl TryFrom<ProtoUuid> for Uuid {
+    type Error = UuidError;
+    fn try_from(uuid: ProtoUuid) -> Result<Self, Self::Error> {
+        Uuid::parse_str(&uuid.value)
+    }
+}
+
+/// Scalar
+impl From<ProtoScalar> for Scalar {
+    fn from(scalar: ProtoScalar) -> Self {
+        Scalar::from_be_bytes_mod_order(&scalar.value)
+    }
+}
 
 #[cfg(test)]
 mod tests {
-    use super::NewPeer;
+    use crate::{AddOrderBuilder, AddPeersBuilder, NetworkOrderBuilder, PeerInfoBuilder};
+
+    use super::{AddOrder, AddPeers};
     use prost::Message;
 
     /// Tests the add new peer message
     #[test]
```

Comment:
is the plan to add serde tests for all proto definitions? feels weird to have it just be for some types, but can also just leave the serde code path to be covered by future unit tests of the state module

Reply:
No, at that point we're just testing the library, these are more sanity checks that the `prost` configuration ends up with what we want. No need to over-test